bc_obps/service/compliance_obligation_service.py

```diff
@@ -0,0 +1,130 @@
+from datetime import date
+from decimal import Decimal
+from django.db import transaction
+from django.core.exceptions import ValidationError
+from compliance.models import ComplianceObligation, ComplianceSummary
+
+
+class ComplianceObligationService:
+    """
+    Service for managing compliance obligations
+    """
+
+    @classmethod
+    def create_compliance_obligation(
+        cls, compliance_summary_id: int, emissions_amount: Decimal, reporting_year: int
+    ) -> ComplianceObligation:
+        """
+        Creates a compliance obligation for a compliance summary
+
+        Args:
+            compliance_summary_id (int): The ID of the compliance summary
+            emissions_amount (Decimal): The amount of excess emissions in tCO2e
+            reporting_year (int): The reporting year for calculating the obligation deadline
+
+        Returns:
+            ComplianceObligation: The created compliance obligation
+
+        Raises:
+            ComplianceSummary.DoesNotExist: If the compliance summary doesn't exist
+        """
+        with transaction.atomic():
+            compliance_summary = ComplianceSummary.objects.get(id=compliance_summary_id)
+
+            # Set obligation deadline to November 30 of the following year
+            # per section 19(1)(b) of BC Greenhouse Gas Emission Reporting Regulation
+            obligation_deadline = cls.get_obligation_deadline(reporting_year)
+
+            obligation = ComplianceObligation.objects.create(
+                compliance_summary=compliance_summary,
+                emissions_amount_tco2e=emissions_amount,
+                status=ComplianceObligation.ObligationStatus.OBLIGATION_NOT_MET,
+                penalty_status=ComplianceObligation.PenaltyStatus.NONE,
+                obligation_deadline=obligation_deadline,
+            )
+
+            return obligation
+
+    @classmethod
+    def get_obligation_deadline(cls, year: int) -> date:
+        """
+        Gets the excess emissions obligation deadline for a year (November 30 of following year)
+
+        Args:
+            year (int): The compliance period year
+
+        Returns:
+            date: The excess emissions obligation deadline (November 30 of following year)
+        """
+        # Per section 19(1)(b) of BC Greenhouse Gas Emission Reporting Regulation
+        return date(year + 1, 11, 30)
```

Comment:
Flagging that without timezone information, this might have erratic behaviour +/- 9 hours